gh-91928: Add `datetime.UTC` alias for `datetime.timezone.utc`

Kab1r opened the pull request:
### fixes #91928

`UTC` is now module attribute aliased to `datetime.timezone.utc`.
You can now do the following:
```python
from datetime import UTC
```
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->

Automerge-Triggered-By: GH:pganssle

Kab1r committed:
feat(datetime): Add `datetime.UTC` alias for `datetime.timezone.utc`

fixes #91928

Kab1r committed:
docs: Add NEWS entry for #91973

pganssle commented on Misc/NEWS.d/next/Library/2022-04-26-18-02-44.gh-issue-91928.V0YveU.rst:
```diff
@@ -0,0 +1 @@
+Add `datetime.UTC` alias for `datetime.timezone.utc`.
```
Do you want to add "Patch by Kabir Kwatra" here? Up to you, but many people do it.

pganssle commented on Doc/library/datetime.rst:
```diff
@@ -84,6 +84,10 @@ The :mod:`datetime` module exports the following constants:
 
+.. attribute:: UTC
+
+   alias for the UTC timezone, ``datetime.utc``.
```
Tweak to this:

```suggestion
   Alias for the UTC timezone singleton :attr:`datetime.timezone.utc`.
   
   .. versionadded:: 3.11
```

pganssle commented on Doc/library/datetime.rst:
```diff
@@ -84,6 +84,10 @@ The :mod:`datetime` module exports the following constants:
    The largest year number allowed in a :class:`date` or :class:`.datetime` object.
    :const:`MAXYEAR` is ``9999``.
 
+.. attribute:: UTC
```
I think this is maybe supposed to be `data` like the ones above it?

Kab1r replied:
I looked at other modules, and it seemed that attribute would fit better here since `UTC` is not technically a constant.

pganssle replied:
OK, that's fine.

pganssle reviewed (changes requested):
This looks great, thanks!

In addition to my other comments, two other things:

1. As I believe you are a first-time contributor, you should probably add your name to [Misc/ACKS](https://github.com/python/cpython/blob/f60b4c3d74f241775f80affe60dcba6448634fe3/Misc/ACKS).
2. Maybe add a very simple test to `TestModule`, like this:

```python
def test_utc_alias(self):
    self.assertIs(UTC, timezone.utc)
```

Thanks for your contribution!

Kab1r committed:
docs: add contribution acknowledgement

Kab1r committed:
docs(datetime): clarify `UTC` timezone singleton

pganssle commented on Doc/library/datetime.rst:
```diff
@@ -84,6 +84,10 @@ The :mod:`datetime` module exports the following constants:
+.. attribute:: UTC
+
+   Alias for the UTC timezone singleton :attr:`datetime.timezone.utc`.
+
```
Still needs `.. versionadded:: 3.11`.

Kab1r replied:
Added!

pganssle reviewed (changes requested):
Can you add this test?

> Maybe add a very simple test to TestModule, like this:

> ```python
>     def test_utc_alias(self):
>        self.assertIs(UTC, timezone.utc)
>```

Kab1r committed:
docs(datetime): add versionadded to `datetime.UTC`

Kab1r committed:
test(datetime): add test for `datetime.UTC` alias

Kab1r committed:
fix(datetime): move `datetime.UTC` definition to end to ensure that it is an alias

pganssle committed:
Clean up nits

1. Add UTC definition into C implementation as well (just for
   consistency, there's no particular reason not to simply assign the
   alias in the Python code later).

2. Move `..versionadded` tag into the right place
